CMakeLists.txt

```diff
@@ -145,6 +145,8 @@ target_include_directories(kimera_vio
   PUBLIC
     ${OpenCV_INCLUDE_DIRS}
     ${GTSAM_INCLUDE_DIR}
+    ${DBoW2_INCLUDE_DIRS}/..
+    ${opengv_DIR}/../../../include
```

Comment:
Why is this needed? I would actually even argue that none of these should be needed as long as target_link_libraries is complete (this should set the include_directories automatically unless the dependencies are not exporting their includes correctly)

Reply:
> unless the dependencies are not exporting their includes correctly

There are a number of known issues back in catkin the would hide incorrect installation rules of headers given the melting pot that was the `devel` space. So some folks never needed to write correct install rules for exporting library artifacts. Now with colcon as the latest build tool for ROS workspaces, one needs to be more rigorous and explicit in writing install rules.

I'm not a CMake guru so perhaps you might spot a better fix here or in upstream packages:
* https://github.com/dorian3d/DBoW2/pull/55
* https://github.com/laurentkneip/opengv/pull/104

Just be sure to test it using a proper isolated cmake build workspace, like colcon does in the included Dockerfile.

Reply:
Lets move this to a new PR, I'd just like to get the rosdistro keys resolved.